File: solutions/c/prime-factors/2/prime_factors.c
#include "prime_factors.h"

size_t find_factors(uint64_t n, uint64_t factors[static MAXFACTORS]){
    size_t idx=0;
    for(uint64_t k=2; k<=n/2; k++)
            while(!(n % k)){
                factors[idx] = k;
                idx++;
                n /= k;
                }
        if (n>1) {
            factors[idx] = n;
            idx++;
        }
        return idx;
       }
    

File: solutions/julia/prime-factors/2/prime-factors.jl
function prime_factors(num)
    factors = []
    while num % 2 == 0
        push!(factors, 2)
        num = div(num,2)
    end
    for k in 3:2:isqrt(num)
       while num % k == 0 
            push!(factors, k)
            num = div(num,k)
        end
    end
    num>1 && push!(factors,num)
    factors
end
